packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -0,0 +1,521 @@
+      return bytecode;
+    } catch (err) {
+      // Catch to try the next RPC
+      console.log(err);
+    }
+  }
+  throw new Error('None of the RPCs responded');
+}
+
+async function getTx(creatorTxHash: string, sourcifyChain: SourcifyChain) {
+  if (!sourcifyChain?.rpc.length)
+    throw new Error('No RPC provider was given for this chain.');
+  for (const rpcURL of sourcifyChain.rpc) {
+    try {
+      const web3 = rpcURL.startsWith('http')
+        ? new Web3(new Web3.providers.HttpProvider(rpcURL))
+        : new Web3(new Web3.providers.WebsocketProvider(rpcURL));
+
+      if (!web3.currentProvider) throw new Error('No provider found');
+
+      // Race the RPC call with a timeout
+      const tx = (await Promise.race([
+        web3.eth.getTransaction(creatorTxHash),
+        rejectInMs(RPC_TIMEOUT, rpcURL),
+      ])) as Transaction;
+      if (tx) {
+        console.log(`Transaction ${creatorTxHash} fetched via ${rpcURL}`);
+      }
+      return tx;
+    } catch (err) {
+      // Catch to try the next RPC
+      console.log(err);
+    }
+  }
+  throw new Error('None of the RPCs responded');
+}
+
+const rejectInMs = (ms: number, host: string) =>
+  new Promise<string>((_resolve, reject) => {
+    setTimeout(() => reject(`RPC ${host} took too long to respond`), ms);
+  });
+
+export function addLibraryAddresses(
+  template: string,
+  real: string
+): {
+  replaced: string;
+  libraryMap: StringMap;
+} {
+  const PLACEHOLDER_START = '__$';
+  const PLACEHOLDER_LENGTH = 40;
+
+  const libraryMap: StringMap = {};
+
+  let index = template.indexOf(PLACEHOLDER_START);
+  for (; index !== -1; index = template.indexOf(PLACEHOLDER_START)) {
+    const placeholder = template.slice(index, index + PLACEHOLDER_LENGTH);
+    const address = real.slice(index, index + PLACEHOLDER_LENGTH);
+    libraryMap[placeholder] = address;
+    const regexCompatiblePlaceholder = placeholder
+      .replace('__$', '__\\$')
+      .replace('$__', '\\$__');
+    const regex = RegExp(regexCompatiblePlaceholder, 'g');
+    template = template.replace(regex, address);
+  }
+
+  return {
+    replaced: template,
+    libraryMap,
+  };
+}
+
+function extractAbiEncodedConstructorArguments(
+  onchainCreationBytecode: string,
+  compiledCreationBytecode: string
+) {
+  if (onchainCreationBytecode.length === compiledCreationBytecode.length)
+    return undefined;
+
+  const startIndex = onchainCreationBytecode.indexOf(compiledCreationBytecode);
+  return (
+    '0x' +
+    onchainCreationBytecode.slice(startIndex + compiledCreationBytecode.length)
+  );
+}
+
+// TODO move this function somewhere else
+// function d2h(d: number) {
+//   const h = d.toString(16);
+//   return h.length % 2 ? '0' + h : h;
+// }
+
+/**
+ * Calculates the address of the contract created with the CREATE opcode using `tx.from` and `tx.nonce`.
+ *
+ * @param Transaction - creatorTx
+ * @returns string -  calculated address
+ */
+// function calculateCreateAddress(creatorTx: Transaction) {
```

Comment:
Why you commented this?

Reply:
Because I started using ethers.js implementation `getContractAddress` instead of our own implementation as below. 

https://github.com/ethereum/sourcify/blob/cd658ab2fe4c1489e63be31233d03c63852de181/packages/lib-sourcify/src/lib/verification.ts#L260-L268

Seems I forgot to remove this bit of code. Thanks